plugins/woocommerce/src/Internal/RestApi/Routes/V4/Fulfillments/Controller.php

```diff
@@ -722,4 +735,25 @@ private function prepare_error_response( $code, $message, $data ): WP_REST_Respo
 			$data['status'] ?? WP_Http::BAD_REQUEST
 		);
 	}
+
+	/**
+	 * Get all shipping providers.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response
+	 */
+	public function get_providers( WP_REST_Request $request ): WP_REST_Response {
+		$providers = \Automattic\WooCommerce\Internal\Fulfillments\FulfillmentUtils::get_shipping_providers_object();
+		return new WP_REST_Response( $providers, WP_Http::OK );
```

Comment:
I believe we should be using the `prepare_item_for_response` for formatting the response of our V4 APIs. It will handle the coherence to the schema and throw the response codes. The method is defined in the `AbstractController`.

Reply:
Refs
1) [Separated Schema Classes](https://wooengineering.wordpress.com/2025/09/22/developing-rest-api-v4-routes/#3-centralised-error-handling)
2) [Centralised Error Handling](https://wooengineering.wordpress.com/2025/09/22/developing-rest-api-v4-routes/#3-centralised-error-handling)